test/common/config/registry_test.cc

```diff
@@ -103,6 +103,19 @@ TEST(RegistryTest, DEPRECATED_FEATURE_TEST(WithDeprecatedFactoryPublished)) {
                           ->name());
 }
 
+class TestWithDeprecatedPublishedFactoryNoDeprecatedNames : public PublishedFactory {
```

Comment:
nit: you may want to use a more descriptive name for this class, like NoNamePublishedFactory